packages/hooks/src/useAccountsList.ts

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react';
+import useAuthorize from './useAuthorize';
+import useBalance from './useBalance';
+import useCurrencyConfig from './useCurrencyConfig';
+
+/** A custom hook that gets the list of all accounts for the current user. */
+const useAccountsList = () => {
+    const { data: authorize_data, ...rest } = useAuthorize();
+    const { data: balance_data } = useBalance();
+    const { getConfig } = useCurrencyConfig();
+
+    // Add additional information to each account.
+    const modified_accounts = useMemo(() => {
+        return authorize_data.account_list?.map(account => {
+            return {
+                ...account,
+                /** Creation time of the account. */
+                created_at: account.created_at ? new Date(account.created_at) : undefined,
+                /** Date till client has excluded him/herself from the website, only present if client is self excluded. */
+                excluded_until: account.excluded_until ? new Date(account.excluded_until) : undefined,
+                /** Indicating whether the wallet is the currently active account. */
+                is_active: account.loginid === authorize_data.loginid,
+                /** indicating whether the account is a virtual-money account. */
+                is_virtual: Boolean(account.is_virtual),
+                /** indicating whether the account is marked as disabled or not. */
+                is_disabled: Boolean(account.is_disabled),
+                /** indicating whether the account is a trading account. */
+                is_trading: account.account_category === 'trading',
+                /** indicating whether the account is a wallet account. */
+                is_wallet: account.account_category === 'wallet',
+                /** The account ID of specified account. */
+                loginid: `${account.loginid}`,
+                /** Account's currency config information */
+                currency_config: account.currency ? getConfig(account.currency) : undefined,
```

Comment:
@sergei-deriv You can add balance to the `modified_accounts` instead of looping through it allover again and defining new variable
```suggestion
                currency_config: account.currency ? getConfig(account.currency) : undefined,
                balance: balance_data?.accounts?.[account.loginid]?.balance || 0,
```

Reply:
@mahdiyeh-deriv I did this because we have another request for `balance` and in the future when we have `BalanceProvider`, balance will be changed a lot of times and with this `accounts_with_balance` variable we will not need to recalculate `modified_accounts` if just the `balance` changed.